dotnet-desktop-guide/net/winforms/compiler-messages/wfo5001.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Compiler Error WFO5001
+description: Learn more about compiler error WFO5001. ColorMode is for evaluation purposes only and subject to change.
+ms.date: 11/19/2024
+f1_keywords:
+  - "WFO5001"
+helpviewer_keywords:
+  - "WFO5001"
+---
+
+# Compiler Error WFO5001
+
+'System.Windows.Forms.Application.SetColorMode(System.Windows.Forms.SystemColorMode)' is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+
+-or-
+
+'System.Windows.Forms.SystemColorMode' is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+
+The color mode feature is currently experimental and subject to change. This error is generated so that you understand the implications of writing code that sets the color mode of the Windows Forms project. The error must be suppressed to continue. For more information about this API, see [Dark mode](../whats-new/net90.md#dark-mode).
+
+## Example
+
+The following sample generates WFO5001:
+
+```csharp
+namespace MyExampleProject;
+
+static class Program
+{
+    [STAThread]
+    static void Main()
+    {
+        ApplicationConfiguration.Initialize();
+        Application.SetColorMode(SystemColorMode.Dark);
+        Application.Run(new Form1());
+    }    
+}
+```
+
+## To correct this error
+
+Suppress the error and enable access to the API by either of the following methods:
+
+- Add the following `PropertyGroup` to your project file to suppress the error:
+
+  ```xml
+  <PropertyGroup>
+      <NoWarn>$(NoWarn);WFO5001</NoWarn>
+  </PropertyGroup>
+  ```
+
+- Suppress the error in code with the `#pragma warning disable WFO5001` directive:
+
+  ```csharp
+  namespace MyExampleProject;
+  
+  static class Program
+  {
+      [STAThread]
+      static void Main()
+      {
+          ApplicationConfiguration.Initialize();
+  #pragma warning disable WFO5001
+          Application.SetColorMode(SystemColorMode.Dark);
+  #pragma warning restore WFO5001
+          Application.Run(new Form1());
```

Comment:
Note, that if you change the `SystemColorMode` to `DarkMode` (or to `System`, when the System is in dark mode), that then the `SystemColors` also change to a dark mode-based palette.